### PR TITLE
Fix for #3789, separate updates into storage backends individually

### LINKED ISSUE #3789: RuntimeError: INTERNAL Error: INTERNAL Error: Could not find node in column segment tree
#### What happens?
Inserting and deleting same rows in transaction give 'INTERNAL Error' (seems related to https://github.com/duckdb/duckdb/issues/1091 ). This error does not occur when I only execute "Step 2"

#### To Reproduce
```python
con = duckdb.connect("test.duckdb")
con.execute("""CREATE TABLE table1 (
          column1 varchar(36) NOT NULL,
          column2 varchar(250) NOT NULL PRIMARY KEY,
        );""")

# Step 1: insert data
con.execute(
    "INSERT INTO table1(column1, column2) values(?, ?)",
    [
        "bla",
        "1",
    ],
)
con.close()

con = duckdb.connect("test.duckdb")
con.begin()
# Step 2: insert and remove data in transaction
con.execute(
    "INSERT INTO table1(column1, column2) values(?, ?)",
    [
        "bla",
        "2",
    ],
)
con.execute(
    "DELETE FROM table1 WHERE column1 = ?",
    ["bla"],
)
```

```RuntimeError: INTERNAL Error: INTERNAL Error: Could not find node in column segment tree!```

#### Environment (please complete the following information):
 - OS: macOS Montery M1
 - DuckDB Version: 0.3.3 / 0.3.4 / 0.3.5
 - DuckDB Client: Python

#### Before Submitting

- [x] **Have you tried this on the latest `master` branch?**
* **Python**: `pip install duckdb --upgrade --pre`
* **R**: `install.packages("https://github.com/duckdb/duckdb/releases/download/master-builds/duckdb_r_src.tar.gz", repos = NULL)`
* **Other Platforms**: You can find binaries [here](https://github.com/duckdb/duckdb/releases/tag/master-builds) or compile from source.

- [x] **Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?**

### PR DESCRIPTION
**Pull request by hannes**

### CODE REVIEW

**Comment by hannes** [comment]
Failure is unrelated

**Comment by hannes** [comment]
@Maxxen can you please have a look?

**Comment by Maxxen** [comment]
I'm maybe not that qualified to review this part of the code but it seems to me that the core of this change is slicing the updated rowid's instead of assuming they are related consecutively, which seems like it should have caused way more issues previously? One small nit is maybe the formatting in the test, otherwise LGTM.

File: test/sql/storage/issue3789_node_segment_tree.test [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# name: test/sql/storage/issue3789_node_segment_tree.test
+# description: Test for issue #3789, INTERNAL Error: Could not find node in column segment tree
+# group: [storage]
+
+statement ok
+CREATE TABLE table1 (column1 integer, column2 integer,
+        )
```
Comment (Maxxen) [inline test/sql/storage/issue3789_node_segment_tree.test:7]: nit: format

**Review by Maxxen — APPROVED** [review]

**Comment by hannes** [comment]
Failing check is unrelated

**Pull request merged**